packages/utils/src/cmds/transform_translate.js

```diff
@@ -0,0 +1,381 @@
+/**
+ * Example usage:
+ *
+ * d2-utils transform_translate \
+ *   --in-dir ~/development/dhis2/maintenance-app/src/i18n \
+ *   --out-dir ~/development/dhis2/project-doom/i18n \
+ *   --override-existing-files \
+ *   --app-name maintenance-app \
+ *   --languages fr,ar \
+ *   --delete-old-files \
+ *   --log-missing-keys
+ */
+const path = require('path')
+const fs = require('fs')
+const log = require('@dhis2/cli-helpers-engine').reporter
+
+const CONSUMING_ROOT = path.join(process.cwd())
+const TRANSLATION_IN_DIR = path.join(CONSUMING_ROOT, 'src/i18n')
+const TRANSLATION_OUT_DIR = path.join(CONSUMING_ROOT, 'i18n')
+
+const fileIsOldTranslationFile = fileName => fileName.match(/\.properties$/)
+
+const appNamePattern = new RegExp('[a-z]+(-[a-z]+)?-app')
+
+/**
+ * \u[a-zA-Z0-9] will be interpreted as unicode characters and need to be escaped.
+ * The escaped sequence will look like this: %5Cu[a-zA-Z0-9]
+ *
+ * The translation needs to be split by whitespaces first in order to create the
+ * correct structure of the new translation
+ */
+const splitTranslation = (originalTranslation, translation) => {
+    const newParts = translation.split(' ').reduce((parts, curSplit) => {
+        const latestPart = parts[parts.length - 1]
+        const latestPartEscaped = escape(latestPart)
+        const curSplitEscaped = escape(curSplit)
+
+        if (
+            parts.length > 0 &&
+            latestPartEscaped.length + curSplitEscaped.length < 77
+        ) {
+            parts[parts.length - 1] += ` ${curSplit}`
+            return parts
+        }
+
+        if (curSplitEscaped.length < 77) {
+            parts.push(curSplit)
+            return parts
+        }
+
+        curSplitEscaped.match(/.{1,76}(?=(%5Cu))?/g).forEach(escapedSplit => {
+            parts.push(unescape(escapedSplit))
+        })
+
+        return parts
+    }, [])
+
+    return newParts
+}
+
+const getTemplateMainLanguage = () => `
+msgid ""
+msgstr ""
+"Project-Id-Version: i18next-conv\\n"
+"MIME-Version: 1.0\\n"
+"Content-Type: text/plain; charset=utf-8\\n"
+"Content-Transfer-Encoding: 8bit\\n"
+"Plural-Forms: nplurals=2; plural=(n != 1)\\n"
+"POT-Creation-Date: 2019-08-13T13:38:22.134Z\\n"
+"PO-Revision-Date: 2019-08-13T13:38:22.134Z\\n"
+`
+
+const getTemplateAlternativeLanguage = (appName, language) => `
```

Comment:
It would be nice to extract these templates to different files as well, you could just keep them as raw files and read them from the filesystem with something like `fs.readFileSync(path.join(__dirname, 'templatfile.txt'))`